pages/news/2020-XX-XX-new-in-2-3-rekordbox-support.html

```diff
@@ -0,0 +1,58 @@
+
+We probably didn't get every case perfect, so we have added some buttons by the beatgrid editing buttons to shift all cue points in a track by 10 ms (left click) or 1 ms (right click) so you can correct the timing yourself. This only shifts cue points, not the beatgrid (because the same problem happens with Serato and Mixxx does not yet import the beatgrid from Serato). If you want to shift the beatgrid too, you can shift the cue points, then go to a cue point and press the beatgrid shift button.
+
+<div style="text-align: center;"><img src="{% static '/static/images/news/mixxx-rekordbox-usb.png' %}" /></div>
+
+## Can I prepare a Rekordbox removable device from Mixxx to use in a CDJ?
+
+Sadly no, however this would be super cool! If you are so inclined to look at how to implement this and contribute back to Mixxx, it would be most welcome. A great to place to start would be looking into [Kaitai Struct serialization](https://doc.kaitai.io/faq.html#writing) which is a prerequisite for writing the [PDB file format](https://github.com/Deep-Symmetry/crate-digger/blob/master/src/main/kaitai/rekordbox_pdb.ksy).
+
+## Shoutouts
+
+As is often the case in open-source, the development of this new functionality was not possible without input from others. The new Mixxx Rekordbox removable device library feature builds upon the hard work already completed by [Deep Symmetry](https://github.com/Deep-Symmetry) for the [Crate Digger](https://github.com/Deep-Symmetry/crate-digger) library and [Pedro Estrela's](https://github.com/pestrela) [research about MP3 timing shifts](https://github.com/digital-dj-tools/dj-data-converter/issues/3) for [DJ Data Converter](https://github.com/digital-dj-tools/dj-data-converter). Kudos and keep up the great work!
+
+## Future work: CDJs as Mixxx controllers
+
+On a related note, we have [started reverse engineering the HID protocol for the Pioneer CDJ 2000 NXS2](https://mixxx.zulipchat.com/#narrow/stream/113295-controller-mapping/topic/Pioneer.20CDJ-2000NXS2). In future versions of Mixxx this would allow you to plug your laptop using Mixxx into CDJs via a USB cable and use your Mixxx library with the CDJs without needing Rekordbox at all. With the release of Rekordbox 6.0, this feature is now restricted in Rekordbox to customers paying a subscription fee, but Mixxx is free{# TODO: Add link to don't have to pay blogpost #} and unlike Rekordbox, Mixxx runs on Linux. You could even add another controller to a CDJ setup to control other Mixxx features such as an [Allen & Heath Xone K2](https://mixxx.org/wiki/doku.php/allen_heath_xone_k2) for full control of Mixxx's effects while using CDJs!
```

Comment:
```suggestion
On a related note, we have [started reverse engineering the HID protocol for the Pioneer CDJ 2000 NXS2](https://mixxx.zulipchat.com/#narrow/stream/113295-controller-mapping/topic/Pioneer.20CDJ-2000NXS2). In future versions of Mixxx this would allow you to plug your laptop using Mixxx into CDJs via a USB cable and use your Mixxx library with the CDJs without needing Rekordbox at all. With the release of Rekordbox 6.0, this feature is now restricted in Rekordbox to customers paying a subscription fee, but Mixxx is free{# TODO: Add link to don't have to pay blogpost #} and unlike Rekordbox, Mixxx runs on Linux. You could even add another controller to a CDJ setup to control other Mixxx features such as an [Allen & Heath Xone K2](https://github.com/mixxxdj/mixxx/wiki/Allen-&-Heath-Xone-K2-K1) for full control of Mixxx's effects while using CDJs!
```

Link changed since we moved to GitHub wiki